src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1947 @@
+
+            if mam_classification == 'mam_by_muac_only':
+                if whz == '-3<=WHZ<-2':
+                    recov_opt = f"mod_SAM_{recov_how}_recov_to_MAM"
+                else: # whz == 'WHZ<-3':
+                    recov_opt = f"sev_SAM_{recov_how}_recov_to_MAM"
+                age_group = self.module.age_grps.get(df.loc[person_id].age_years, '5+y')
+                wasted_days = (self.sim.date - df.at[person_id, 'un_last_wasting_date_of_onset']).days
+
+                def get_min_length(in_recov_how, in_person_id, in_whz):
+                    if in_recov_how == 'tx':
+                        if df.at[in_person_id, 'un_sam_with_complications']:
+                            return (p['tx_length_weeks_InpatientSAM'] * 7) - 1
+                        else:  # SAM without complications
+                            return (p['tx_length_weeks_OutpatientSAM'] * 7) - 1
+                    else: # in_recov_how == 'nat'
+                        if in_whz == '-3<=WHZ<-2':
+                            return p['duration_of_untreated_mod_wasting'] - 1
+                        else:  # in_whz == 'WHZ<=-3'
+                            return p['duration_of_untreated_mod_wasting'] + p['duration_of_untreated_sev_wasting'] - 2
+
+                assert wasted_days >= get_min_length(recov_how, person_id, whz), \
+                    (f" The {person_id=} is wasted {wasted_days=} < minimal expected length= "
+                     f"{get_min_length(recov_how, person_id, whz)} days when {recov_opt=}.")
+                self.module.wasting_length_tracker[age_group][recov_opt].append(wasted_days)
+
+                df.at[person_id, 'un_WHZ_category'] = 'WHZ>=-2'
+                df.at[person_id, 'un_am_MUAC_category'] = '[115-125)mm'
+
+            if mam_classification == 'mam_by_muac_and_whz':
+                df.at[person_id, 'un_WHZ_category'] = '-3<=WHZ<-2'
+                df.at[person_id, 'un_am_MUAC_category'] = '[115-125)mm'
+
+            if mam_classification == 'mam_by_whz_only':
+                df.at[person_id, 'un_WHZ_category'] = '-3<=WHZ<-2'
+                df.at[person_id, 'un_am_MUAC_category'] = '>=125mm'
+
+        # Update all other properties equally
+        df.at[person_id, 'un_clinical_acute_malnutrition'] = 'MAM'
+        df.at[person_id, 'un_am_nutritional_oedema'] = False
+        df.at[person_id, 'un_sam_with_complications'] = False
+        df.at[person_id, 'un_sam_death_date'] = pd.NaT # death is cancelled if was scheduled
+        df.at[person_id, 'un_am_tx_start_date'] = pd.NaT
+        # Start without treatment, treatment will be applied with HSI if care sought
+        df.at[person_id, 'un_am_treatment_type'] = 'none'
+
+        # this will clear all wasting symptoms (applicable for SAM, not MAM)
+        self.sim.modules["SymptomManager"].clear_symptoms(
+            person_id=person_id, disease_module=self.module
```

Comment:
why not refer to the symptom self.module.symptom_of_wasting (quicker than asking symptom manager to do a lookup)